src/action/base/create_or_merge_nix_config.rs

```diff
@@ -0,0 +1,490 @@
+use std::{
+    os::unix::fs::PermissionsExt,
+    path::{Path, PathBuf},
+};
+
+use nix_config_parser::NixConfig;
+use rand::Rng;
+use tokio::{
+    fs::{remove_file, OpenOptions},
+    io::AsyncWriteExt,
+};
+use tracing::{span, Span};
+
+use crate::action::{Action, ActionDescription, ActionError, StatefulAction};
+
+/// The `nix.conf` configuration names that are safe to merge.
+// FIXME(@cole-h): make configurable by downstream users?
```

Comment:
I don't know if this should be a blocker for this PR, but it's definitely something I want to do in a future PR.

The API / "how" of it is still very much up in the air either way, though.

Reply:
I think this is fine for now. :)